### PR TITLE
Add the ability to demangle java resource files

### PR DESCRIPTION
**Pull request by miguelsaldivar**
This is a patch that progresses towards getting the libiberty tests to pass. The corresponding libiberty patch can be seen [here](https://gcc.gnu.org/legacy-ml/java-patches/2008-q1/msg00029.html). 

There might be a better way of demangling a `ResourceName`, but I couldn't really think of an obvious solution, please let me know if there's a better way.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -136,7 +136,7 @@ use std::fmt::Write;
         s.extend(87..89);
         s.extend(91..93);
         s.extend(94..105);
-        s.extend(106..113);
+        s.extend(105..113);
```
Comment (khuey) [inline build.rs:139]: This can be consolidated with the line above it.

File: src/ast.rs [hunk 2]
```diff
@@ -7357,6 +7361,27 @@ impl Parse for SpecialName {
                 };
                 Ok((SpecialName::GuardTemporary(name, idx), tail))
             }
+            b"Gr" => {
+                let (resource_name_len, tail) = parse_number(10, false, tail)?;
+                debug_assert!(resource_name_len >= 0);
```
Comment (khuey) [inline src/ast.rs:7366]: Invalid input could give you a negative number here, no?

File: build.rs (hunk 1 above)
Comment (miguelsaldivar) [reply build.rs:139]: Dooh, thanks for catching that.

File: src/ast.rs (hunk 2 above)
Comment (miguelsaldivar) [reply src/ast.rs:7366]: I tried breaking it, and couldn't :(. Can you think of any invalid input that would break it?

Comment (khuey) [reply src/ast.rs:7366]: Ah you are passing false for `allow_signed` so I suppose I'm wrong.

(Also means you can leave the assert off, I think).

Comment (miguelsaldivar) [reply src/ast.rs:7366]: Yup, that's redundant let me remove that.

**Comment by khuey** [comment]
Thanks!

**Pull request merged**